### ISSUE TITLE
Readme: Add example of how to actually run sparkle-example-lda on EMR

### ISSUE THREAD

**Issue by nh2** [tweag/sparkle#48]
I found it quite a hurdle to actually get one of the examples running on EMR.

It would be nice if the README had step-by-step instructions on how to do that.

On my way to getting it to work, I found at least the following information missing:

* How exactly to build jars from the examples (`stack --nix build lda && stack --nix exec -- sparkle package sparkle-example-lda`)
* How to launch them in EMR (`scp` the jar up and run it from the EMR master using e.g.  `spark-submit ./sparkle-example-lda.jar`)
* What region to create your EMR cluster in (see #47; I had to copy up the files into my own S3 bucket accessible from my region by hand)
* That sparkle apparently only works with Spark 1.6, not with 2.0, but 2.0 is the current default on Amazon EMR
* That for getting the option to Spark 1.6 cluster you have to choose `Release` `emr-4.7.2` when creating a cluster
* That when you do that, and running above-mentioned `spark-submit`, you'll get `Exception in thread "main" java.lang.UnsupportedClassVersionError: io/tweag/sparkle/SparkMain : Unsupported major.minor version 52.0` because Nix and gradle are configured to use Java 8, but EMR's Spark only supports Java 7
* That you can fix that by adding

    ```
    compileJava {
      sourceCompatibility = 1.7
      targetCompatibility = 1.7
    }
    ```

    to `build.gradle` and changing `openjdk` to `openjdk7` in `shell.nix` (I'm on commit ccd36135d).

**Comment by mboes** [comment]
Regarding the first point, I believe the existing README covers this already. Right at the top of the README there is a tl;dr section saying

```
$ stack --nix build hello
$ stack --nix exec sparkle package sparkle-example-hello
```

and then the rest explains what each command does. So I'm at a loss regarding how to improve this. Maybe send a PR if you have an idea?

Furthermore, how to work with EMR was the subject of this blog post: http://blog.tweag.io/posts/2016-06-20-haskell-compute-paas-with-sparkle.html. The README now links to it directly. Note the screenshot included there, which specifies emr-4.7.1 and Spark 1.6.

Things inevitably break over time as external resources (Spark documentation URL's, default versions on EMR) move forward to newer versions. I just pushed 211029cdc58cafe587c35899d2814819b9703f2b to pin Nix to openjdk7, as you suggest. This will avoid issues with the `openjdk` attribute being remapped to a different derivation in the future in the upstream Nixpkgs repo. With this patch, I believe the last few steps in your description are now redundant.

**Comment by nh2** [comment]
> Right at the top of the README

Oops, you're right

> Furthermore, how to work with EMR was the subject of this blog post: http://blog.tweag.io/posts/2016-06-20-haskell-compute-paas-with-sparkle.html

Ah, great. I had confused it with the older one http://www.tweag.io/blog/haskell-meets-large-scale-distributed-analytics -- the one you link contains the infos I needed.

> With this patch, I believe the last few steps in your description are now redundant.

Yep, thank you!
